### PR TITLE
[flang] use setsid to assign the child to prevent zombie as it will be clean up by init process

### PR DESCRIPTION
**Pull request by yiwu0b11**
When using `setsid()` in a child process created by `fork()`, a new session is created, and the child becomes a session leader. If the parent process terminates before the child, the child becomes an orphan and is adopted by the `init` process. The `init` process will eventually clean up the child process once it exits. 

However, killing the parent does not automatically kill the child; the child will continue running until it exits. 
Proper cleanup involves waiting for the child process to exit using `wait()` or `waitpid()` in the parent process to avoid zombie processes, but this approach is not valid for `EXECUTE_COMMAND_LINE` with async mode.

### CODE REVIEW

**Comment by yi-wu-arm** [comment]
when forcing printout from child and parent:
```
$ ./a.out 
Parent process (PID: 273500) is running in a new session (SID: 2347).
Child process (PID: 273501) is running in a new session (SID: 273501).
```
A different session id is what we want.

File: flang/runtime/execute.cpp [hunk 1]
```diff
@@ -191,6 +189,19 @@ void RTNAME(ExecuteCommandLine)(const Descriptor &command, bool wait,
         CheckAndCopyCharsToDescriptor(cmdmsg, "Fork failed");
       }
     } else if (pid == 0) {
+      // Create a new session, let init process take care of zombie child
+      if (setsid() == -1) {
+        if (!cmdstat) {
+          terminator.Crash("setsid() failed with errno: %d, asynchronous "
+                           "process initiation failed.",
+              errno);
```
Comment (DavidTruby) [inline flang/runtime/execute.cpp:197]: I guess this formatting comes from

**Review by DavidTruby — APPROVED** [review]
LGTM but give a bit of time for anyone else to have a look in case my understanding (which matches yours) is wrong.

**Review by psteinfeld — COMMENTED** [review]
What's the purpose of this change?  Is there a test that's failing?

Also, I don't see the changes listed here, but when I try to build from this repository, I get errors.  The function `ExecuteCommandLine` declares a variable called `newCmd`.  The version of that declaration that I see in this repository looks like this:
```
    const char *newCmd{EnsureNullTerminated(
     ...
```
Then, later in this function this variable is deallocated.  For me, this causes the following error message:
```
/local/home/psteinfeld/main/yiwu/llvm-project/flang/runtime/execute.cpp:213:24: error: cast from type ‘const char*’ to type ‘void*’ casts away qualifiers [-Werror=cast-qual]
  213 |     FreeMemory((void *)newCmd);
      |                        ^~~~~~
At global scope:
cc1plus: error: unrecognized command line option ‘-Wno-ctad-maybe-unsupported’ [-Werror]
cc1plus: all warnings being treated as errors
```
```

**Review by psteinfeld — APPROVED** [review]
All builds and tests correctly and looks good.

**Comment by kiranchandramohan** [comment]
Could you add a test?

**Review by kiranchandramohan — APPROVED** [review]
LG.

**Pull request merged**